modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -310,94 +276,58 @@ def get_files_info(ip, _rport, shares, info)
     end
   end
 
-  def rport
-    @rport || datastore['RPORT']
-  end
-
-  # Overrides the one in smb.rb
-  def smb_direct
-    @smb_redirect || datastore['SMBDirect']
-  end
-
   def run_host(ip)
-    @rport = datastore['RPORT']
-    @smb_redirect = datastore['SMBDirect']
-    @srvsvc = datastore['USE_SRVSVC_ONLY']
     shares = []
+    begin
+      print_status 'Starting module'
+      connect(versions: [1, 2, 3])
+      smb_login
+      shares = simple.client.net_share_enum_all(ip)
+      os_info = get_os_info(ip)
+      print_status(os_info) if os_info
+
+      if shares.empty?
+        print_status('No shares collected')
+      else
+        shares.each do |share|
+          print_good("#{share[:name]} - (#{share[:type]}) #{share[:comment]}")
+        end
 
-    [[139, false], [445, true]].each do |info|
-      @rport = info[0]
-      @smb_redirect = info[1]
-
-      begin
-        connect
-        smb_login
-        shares = smb_netshareenumall
-
-        os_info = get_os_info(ip, rport)
-        print_status(os_info) if os_info
-
-        if shares.empty?
-          print_status('No shares collected')
-        else
-          shares_info = shares.map { |x| "#{x[0]} - (#{x[1]}) #{x[2]}" }.join(', ')
-          shares_info.split(', ').each do |share|
-            print_good share
-          end
-          report_note(
-            host: ip,
-            proto: 'tcp',
-            port: rport,
-            type: 'smb.shares',
-            data: { shares: shares },
-            update: :unique_data
-          )
-
-          if datastore['SpiderShares']
-            begin
-              connect(versions: [1])
-              smb_login
-              get_files_info(ip, rport, shares, info)
-            rescue ::Rex::Proto::SMB::Exceptions::Error, Errno::ECONNRESET => e
-              print_error(
-                "Error when Spidering shares recursively (#{e}). This feature "\
-                'is only available with Rex client (SMB1 only) and the host '\
-                "probably doesn't support SMB1."
-              )
-            end
-          end
-
-          break if rport == 139
+        report_note(
+          host: ip,
+          proto: 'tcp',
+          port: rport,
+          type: 'smb.shares',
+          data: { shares: shares }, # We are now storing an array of hashes here, rather than an array of arrays. Will this cause issues further down the line?
```

Comment:
Does this question need investigated, or is this a question to the reader

Reply:
I would like to investigate it so that no additional functionality gets broken because of this change.